From d6350e23dea044118c8b2841aea640d2ca231d00 Mon Sep 17 00:00:00 2001
From: Mingcong Bai <jeffbai@aosc.io>
Date: Thu, 21 Aug 2025 17:02:48 +0800
Subject: [PATCH 326/326] AOSCOS: serial: 8250: 8250_loongson: sanitize
 uart_port::serial_{in,out}() types

Follow commit fc9ceb501e38 ("serial: 8250: sanitize
uart_port::serial_{in,out}() types") and revise serial_{in,out}() types.

Fixes: "FROMLIST: tty: serial: 8250: Add loongson uart driver support"
Signed-off-by: Mingcong Bai <jeffbai@aosc.io>
---
 drivers/tty/serial/8250/8250_loongson.c | 12 +++++-------
 1 file changed, 5 insertions(+), 7 deletions(-)

diff --git a/drivers/tty/serial/8250/8250_loongson.c b/drivers/tty/serial/8250/8250_loongson.c
index 88205fed4fd30..068e3bd021f26 100644
--- a/drivers/tty/serial/8250/8250_loongson.c
+++ b/drivers/tty/serial/8250/8250_loongson.c
@@ -58,19 +58,17 @@ static unsigned int serial_fixup(struct uart_port *p, unsigned int offset, unsig
 	return val;
 }
 
-static unsigned int loongson_serial_in(struct uart_port *p, int offset)
+static unsigned int loongson_serial_in(struct uart_port *p, unsigned int offset)
 {
-	unsigned int val, offset0 = offset;
-
-	offset = offset << p->regshift;
-	val = readb(p->membase + offset);
+	unsigned int offset = offset << p->regshift;
+	u32 val = readb(p->membase + offset);
 
 	return serial_fixup(p, offset0, val);
 }
 
-static void loongson_serial_out(struct uart_port *p, int offset, int value)
+static void loongson_serial_out(struct uart_port *p, unsigned int offset, u32 value)
 {
-	offset = offset << p->regshift;
+	unsigned int offset = offset << p->regshift;
 	writeb(serial_fixup(p, offset, value), p->membase + offset);
 }
 
-- 
2.51.0

